src/middlewared/middlewared/plugins/failover_/event.py

```diff
@@ -588,11 +604,28 @@ def vrrp_backup(self, job, fobj, ifname, event):
         if fw_drop_job.error:
             logger.error(f'Error blocking network traffic: {fw_drop_job.error}')
 
-        # restarting keepalived sends a priority 0 advertisement
-        # which means any VIP that is on this controller will be
-        # migrated to the other controller
-        logger.info('Transitioning all VIPs off this node')
-        self.run_call('service.restart', 'keepalived')
+        start_new_vrrp_thread = False
+        try:
+            for thread in filter(lambda x: x.name == vrrp_events.VRRP_THREAD_NAME, threading.enumerate()):
+                if thread.is_alive():
+                    # we have to stop the thread responsible for processing "failover" events
+                    # because restarting the keepalived triggers a backup event...but we're
+                    # already processing one at this point so this would cause an endless
+                    # backup failover loop...which isn't good
+                    logger.info('Temporarily stopping vrrp event thread')
+                    thread.shutdown()
+                start_new_vrrp_thread = True
+                break
+
+            # restarting keepalived sends a priority 0 advertisement
+            # which means any VIP that is on this controller will be
+            # migrated to the other controller
+            logger.info('Transitioning all VIPs off this node')
+            self.run_call('service.restart', 'keepalived')
+        finally:
+            if start_new_vrrp_thread:
+                logger.info('Restarting vrrp event thread')
+                vrrp_events.VrrpFifoThread(middleware=self.middleware).start()
```

Comment:
Should we have a single way to start/stop vrrp thread? We can even move it to its own `failover.vrrp` plugin with its own start/stop methods.

Reply:
yeah, I thought about that but got lazy. I'll create a separate plugin